src/hooks/useSelector.js

```diff
@@ -51,15 +49,15 @@ export function useSelector(selector, deps) {
     contextSub
   ])
 
-  const memoizedSelector = useMemo(() => selector, deps)
-
   const latestSubscriptionCallbackError = useRef()
-  const latestSelector = useRef(memoizedSelector)
+  const latestSelector = useRef(selector)
 
-  let selectedState = undefined
+  let latestSelectedState = useRef()
 
   try {
-    selectedState = memoizedSelector(store.getState())
+    latestSelectedState.current = useMemo(() => selector(store.getState()), [
```

Comment:
This is completely wrong, since if the selector does not change, it will always have the result based of the state at the point the selector changed.

Reply:
Hi @MrWolfZ I have added an Edit that explains the rationale of this PR a bit better. If you really think that this is wrong, could you please provide a test that proves that?

Reply:
Sure, here you go:

```jsx
it('always uses the latest state', () => {
  store = createStore(c => c + 1, -1)

  const Comp = () => {
    const selector = useCallback(c => c + 1, [])
    const value = useSelector(selector)
    renderedItems.push(value)
    return <div />
  }

  rtl.render(
    <ProviderMock store={store}>
      <Comp />
    </ProviderMock>
  )

  expect(renderedItems).toEqual([1])

  store.dispatch({ type: '' })

  expect(renderedItems).toBe([1, 2])
})
```

Reply:
In general, I really wouldn't worry too much about how often the selector is called. Even if this change worked, it wouldn't be worth the added complexity

Reply:
Ups! You were right! I just updated it the code, I think that I made it a lot simpler (a lot less refs) and more performant... Wit this latest code I think that I haven't added complexity, on the contrary and on top of that we don't have unnecessary evaluations of selector :smile: